lib/proxy.js

```diff
@@ -137,12 +137,13 @@ function buildResultingMiddleware (host, options, stream) {
     var reqOpts = {
       url     :  URL.format(proxyUrl),
       method  : req.method,
-      headers : _.defaults(options.headers || {}, req.headers),
+      // headers : _.defaults(options.headers || {}, req.headers),
+      headers : options.headers,
```

Comment:
I don't think what you had was a bad idea. supertest is sending the header `accept-encoding` with a value of `gzip, deflate` and i think that is causing an issue because the request is then erroring out saying `Uncaught error: incorrect header check`

![image](https://cloud.githubusercontent.com/assets/2179012/6436081/daad708e-c077-11e4-926c-0c23cf8b9f24.png)